evap/static/ts/src/auto-form-saver.ts

```diff
@@ -0,0 +1,198 @@
+                return false;
+            }
+            return true;
+        });
+    }
+
+    getExcludeFields(): Element[] {
+        return this.options.excludeFields.flatMap(selector => Array.from(document.querySelectorAll(selector)));
+    }
+
+    getPrefix(field: Element) {
+        return (
+            this.href +
+            this.target.id +
+            this.target.name +
+            (field.getAttribute("name") ?? "") +
+            this.options.customKeySuffix
+        );
+    }
+
+    bindSaveData() {
+        for (const field of this.findFieldsToProtect()) {
+            if (this.getExcludeFields().includes(field)) {
+                continue;
+            }
+            const prefix = this.getPrefix(field);
+            if ((field instanceof HTMLInputElement && field.type === "text") || field instanceof HTMLTextAreaElement) {
+                this.bindSaveDataImmediately(field, prefix);
+            }
+            this.bindSaveDataOnChange(field);
+        }
+    }
+
+    saveAllData() {
+        for (const field of this.findFieldsToProtect()) {
+            if (this.getExcludeFields().includes(field) || !field.getAttribute("name")) {
+                continue;
+            }
+            const prefix = this.getPrefix(field);
+            const fieldType = field.getAttribute("type");
+            // @ts-expect-error All field objects are some kind of input field with value.
+            let value: string | string[] | boolean = field.value;
+
+            if (field instanceof HTMLInputElement && fieldType === "checkbox") {
+                value = field.checked;
+                this.saveToBrowserStorage(prefix, value, false);
+            } else if (field instanceof HTMLInputElement && fieldType === "radio") {
+                if (field.checked) {
+                    value = field.value;
+                    this.saveToBrowserStorage(prefix, value, false);
+                }
+            } else {
+                this.saveToBrowserStorage(prefix, value, false);
+            }
+        }
+        this.options.onSave(this);
+    }
+
+    restoreAllData() {
+        let restored = false;
+
+        for (const field of this.findFieldsToProtect()) {
+            if (this.getExcludeFields().includes(field)) {
+                continue;
+            }
+            const storedValue = localStorage.getItem(this.getPrefix(field));
+            if (storedValue !== null) {
+                this.restoreFieldsData(field, storedValue);
+                restored = true;
+            }
+        }
+
+        if (restored) {
+            this.options.onRestore(this);
+        }
+    }
+
+    restoreFieldsData(field: Element, storedValue: string) {
+        if (field.getAttribute("name") === undefined) {
+            return false;
+        }
+        if (field instanceof HTMLInputElement && field.type === "checkbox") {
+            field.checked = storedValue !== "false";
```

Comment:
Now that I see it, should we go for `storedValue === "true"` instead? What do we save to localStorage if we have `input type=checkbox checked=1`?